File: two-sum/two-sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # initialise a dictionary
        seen = {}
        
        for index, item in enumerate(nums):
            # calculate the difference
            diff = target - item
            
            if diff in seen:
                # if the difference is in the seen, return it
                return [seen[diff], index]
            else:
                # if not, set it 
                seen[item] = index
                
        return []
    
    

File: two-sum/two-sum.js
// https://leetcode.com/problems/two-sum/

var twoSum = function (nums, target) {
    // initialise a dictionary
    var map = {}

    for (let i = 0; i < nums.length; i++) {
        // calculate the difference
        var diff = target - nums[i]

        if (map[diff] != undefined) {
            // if the difference is in the seen, return it
            return [i, map[diff]]
        } else {
            // if not, set it
            map[nums[i]] = i
        }
    }
};